dask-metrics/dask_metrics/monitor.py

```diff
@@ -0,0 +1,642 @@
+        
+        Parameters:
+        tracking : list[str], optional
+            Sets the list of metrics to be tracked on each worker. Clears
+            stored metrics if the list is being updated
+            > See attach() for metric options
+        """
+        if len(tracking) > 0:
+            # update tracking list if new one passed
+            self.send({
+                'op': 'update_tracking',
+                'tracking': tracking
+            })
+            
+            # new metrics might be incompatible format with current
+            # metrics, so the old ones must be cleared
+            self.metrics = []
+            self.metric_state = MetricState.NONE
+        if len(self.metrics) > 0:
+            self.metric_state = MetricState.HAS_SOME
+        self.send({'op': 'start_recording'}) # start collecting metrics
+        
+    def stop(self, force=False):
+        """
+        Stops the collection of metrics on all workers.
+        """
+        self.send({
+            'op': 'stop_recording',
+            'force': force
+        })
+    
+    def shutdown(self):
+        """
+        Removes the monitor from the cluster and deletes any remaining
+        temporary files that might be left on workers.
+        """
+        self.send({'op': 'metrics_shutdown'})
+        
+    def new_job(self, job=None):
+        """
+        Used to indicate which job is currently executing to the monitor. Only used
+        if job_type is set to 'manual' upon monitor startup.
+        
+        Parameters:
+        job : int, optional
+            If set, the new job number is set to the value passed to job,
+            otherwise the job number is incremented by one.
+        """
+        if self.job_type == 'manual':
+            self.send({
+                'op': 'new_job',
+                'job': job
+            })
+        else:
+            print('Manual job signalling not enabled')
+        
+    def request_metrics(self):
+        """
+        Pulls all metrics collected until now by the workers and stores them in memory.
+        """
+        if self.metric_state == MetricState.HAS_ALL:
+            return
+        
+        def func(data, all_metrics, successful): # callback func to recieve metrics
+            self.metrics = data
+            self.metric_state = MetricState.HAS_ALL if all_metrics else MetricState.HAS_SOME
+            self.successful_jobs = successful
+            self.event.set()
+        
+        self.client._stream_handlers.update({'give_metrics': func})
+        self.send({'op': 'send_metrics'})
+        self.event.wait()
+        self.event.clear()
+    
+    def clear_metrics(self, **kwargs):
+        """
+        Clears the stored metrics on workers and scheduler.
+        """
+        self.metrics = []
+        self.metric_state = MetricState.NONE
+        self.send({'op': 'clear_metrics'})
+        
+    def to_csv(self, path, fname='worker', clear_dir=True):
```

Comment:
I'd prefer defaulting this to `False` since the script clears out all subfolders in the directory by default